src/orion/service/client/workon.py

```diff
@@ -0,0 +1,178 @@
+    def new_experiment(self, name, **config) -> RemoteException:
+        """Create a new experiment"""
+        payload = self._post("experiment", name=name, **config)
+
+        payload.pop("experiment_name", None)
+        payload["name"] = name
+        self.experiment = RemoteExperiment(**payload)
+        return self.experiment
+
+    def suggest(self, pool_size: int = 1, experiment_name=None) -> TrialCM:
+        """Generate a new trial for the current experiment"""
+        experiment_name = experiment_name or self.experiment_name
+
+        if experiment_name is None:
+            raise ExperiementIsNotSetup("experiment_name is not set")
+
+        log.debug("client: suggest: %s", experiment_name)
+        result = self._post(
+            "suggest", experiment_name=experiment_name, pool_size=pool_size
+        )
+
+        trials = []
+        for trial in result["trials"]:
+            trials.append(
+                RemoteTrial(**trial, exp_working_dir=self.experiment.working_dir)
+            )
+
+        # Currently we only return a single trial
+        trial = trials[0]
+
+        # Create pacemaker for them
+        self._maintain_reservation(trial)
+        return TrialCM(self, trial)
+
+    def _maintain_reservation(self, trial):
+        self._pacemakers[trial.id] = TrialPacemaker(trial, self)
+        self._pacemakers[trial.id].start()
+
+    def release(self, trial, status="interrupted"):
+        """Release a trial after work on it finished"""
+        experiment_name = self.experiment_name
+
+        if experiment_name is None:
+            raise ExperiementIsNotSetup("experiment_name is not set")
+
+        failed = False
+        try:
+            self._post(
+                "release",
+                euid=self.experiment_id,
+                experiment_name=experiment_name,
+                trial_hash=trial.id,
+                status=status,
+            )
+        except Exception as e:
+            failed = True
+            raise e
+        finally:
+            pacemaker = self._pacemakers.pop(trial.id, None)
+            if pacemaker is not None:
+                pacemaker.stop()
+            elif not failed:
+                raise RuntimeError(
+                    f"Trial {trial.id} had no pacemakers. Was it reserved properly?"
+                )
+
+    def observe(
+        self,
+        trial: RemoteTrial,
+        results: List[Dict],
+        name: str = "objective",
+        experiment_name=None,
+    ) -> None:
+        """Observe the result of a given trial"""
+        experiment_name = experiment_name or self.experiment_name
+
+        if isinstance(results, numbers.Number):
+            results = [dict(value=results, name=name, type="objective")]
+
+        if experiment_name is None:
+            raise ExperiementIsNotSetup("experiment_name is not set")
+
+        if not isinstance(results, list):
+            raise TypeError("Results need to be a float or a list of values")
+
+        self._post(
```

Comment:
What happens if an exception occurs remotely during observe. Will this `_post` raise a RemoteException?

Reply:
If on the server the ExperimentClient raises an exception, the server convert the exception as a json message.
The message will be read  by the client 

1. if the exception is part of the allowed exception set, then the exact the same exception that was raised by the ExperimentClient remotely is raised locally 
2. if the exception is not allowed (i.e internal server error) a RemoteException is rased 


The setup allow us to reuse the same tests for ExperimentClient and ExperimentClientREST
and provide usage consistency throughout the clients